src/cli/commands/global.js

```diff
@@ -69,44 +72,43 @@ async function getBins(config: Config): Promise<Set<string>> {
   return paths;
 }
 
-function getGlobalPrefix(config: Config, flags: Object): string {
+async function getGlobalPrefix(config: Config, flags: Object): Promise<string> {
   if (flags.prefix) {
     return flags.prefix;
   } else if (config.getOption('prefix')) {
     return String(config.getOption('prefix'));
   } else if (process.env.PREFIX) {
     return process.env.PREFIX;
-  } else if (process.platform === 'win32') {
+  }
+
+  if (process.platform === 'win32') {
+    // %LOCALAPPDATA%\Yarn --> C:\Users\Alice\AppData\Local\Yarn
     if (process.env.LOCALAPPDATA) {
-      return path.join(process.env.LOCALAPPDATA, 'Yarn', 'bin');
-    }
-    // c:\node\node.exe --> prefix=c:\node\
-    return path.dirname(process.execPath);
-  } else {
-    // /usr/local/bin/node --> prefix=/usr/local
-    let prefix = path.dirname(path.dirname(process.execPath));
-
-    // destdir only is respected on Unix
-    if (process.env.DESTDIR) {
-      prefix = path.join(process.env.DESTDIR, prefix);
+      return path.join(process.env.LOCALAPPDATA, 'Yarn');
     }
+    return FALLBACK_GLOBAL_PREFIX;
+  }
 
-    return prefix;
+  try {
+    await fs.access(path.join(POSIX_GLOBAL_PREFIX, 'bin'), nativeFs.constants.W_OK);
```

Comment:
If you want to ensure if we have permission to write to `%LOCALAPPDATA%\Yarn`, that can be done. It's likely redundant though, because it's almost guaranteed to be writable, at least as writable as `~/.yarn` (`FALLBACK_GLOBAL_PREFIX`).

I guess it won't hurt to do that.